src/node/snapshotter.h

```diff
@@ -357,6 +359,8 @@ namespace ccf
       // at the last snapshottable index before idx, and schedule snapshot
       // serialisation on another thread (round-robin). Otherwise, only record
       // that a snapshot was generated.
+
+      kv::ScopedStoreMapsLock maps_lock(store);
```

Comment:
Do we know the place in the code these two locks are acquired in the inverse order, leading to a deadlock?

Reply:
After the fix, I believe there is no more inverse order between map lock and snapshot lock based on TSAN result.

Reply:
I'd be curious too.

Reply:
Please check https://github.com/microsoft/CCF/issues/5051